taskgraph/Task.py

```diff
@@ -1515,6 +1619,8 @@ def _execute_sqlite(
             raise ValueError('Unknown mode: %s' % mode)
 
         if execute == 'execute':
+            if argument_list is None:
+                argument_list = []

```

Comment:
Great question and I've thought about this specific thing more than you might think! The problem with setting a default parameter to `[]` is that the parameter becomes that specific empty list object. If the code base did something to that list (which a good programmer never should) like append a value or something, it modifies the default "empty list" to now have an element in it. So in general it's considered bad practice to set a default argument to `[]`. In turn though, I found myself often forgetting to pass an empty list as that parameter all the time. So this "`if argument_list is None: ... argument_list = []` is a weird bastardization of "good" Python coding practice. I'm also in this headspace because we do this exact thing in the TaskGraph's `__init__` function, except there it makes more sense because we may modify those lists later.

ANYWAY, I see how confusing that is and I've just struck the whole thing and put in an if/else with two different ways to call `.execute` in 18a95d1355932f9fe53a739a64d9a1cfc939ffc9